ci-operator/step-registry/cucushift/installer/rehearse/azure/stack/upi/deprovision/cucushift-installer-rehearse-azure-stack-upi-deprovision-chain.yaml

```diff
@@ -0,0 +1,7 @@
+chain:
+  as: cucushift-installer-rehearse-azure-stack-upi-deprovision
+  steps:
+  - chain: gather
```

Comment:
Compared with `upi-azurestack-post` chain, seem like we should call `chain: ipi-deprovision`

Reply:
I guess that would help destroy the *.apps records.

Reply:
In step [upi-install-azurestack](https://github.com/openshift/release/blob/master/ci-operator/step-registry/upi/install/azurestack/upi-install-azurestack-commands.sh#L161), it creates public zone in cluster RG and all dns records(api/api-int/*.apps) are created in this public zone.  So I think it is enough to only run step `upi-deprovision-azurestack` to destroy upi cluster on ash.

The purpose of changing chain `ipi-deprovision` to `gather` is that [job](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/38484/rehearse-38484-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack/1648977719070822400) failed at the step `ipi-deprovision-deprovision` , because it does not copy ["metadat.json"](https://github.com/openshift/release/blob/master/ci-operator/step-registry/upi/install/azurestack/upi-install-azurestack-commands.sh#L268) at the end in step `upi-install-azurestack-commands.sh` , then result in line 13 in step [ipi-deprovision-deprovision](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh#L13) always fails.

Reply:
Got it. Thanks for the detailed explaination.